### PR TITLE
[ML][Data Frame] Adding bwc tests for pivot transform

### PR DESCRIPTION
**Pull request by benwtrent**
This adds some new BWC tests for data frame pivot transforms.

When backporting, we will have to black list these tests on clusters where the old version is before 7.2.0. 

Additionally, since `pipeline` is not supported in 7.2.0, that field will have to be removed from the `complex` old cluster job. 

Since it is not possible to capture all combinations of parameters efficiently, I opted to have two transforms for the Old, mixed, and new cluster tests. One with minimal viable config, and another with a config that fills out all optional fields.

### CODE REVIEW

**Review by davidkyle — COMMENTED** [review]
Can the test have a DF created in the old cluster and left started during upgrade? That way we check DFs left running survive a rolling upgrade. The assertion would be that task state == STARTED but it would be good to check the DF has an assignment. Can the test index more data in the upgraded cluster and restart the DF then assert the stat counts have updated. If that isn't possible in the yml test framework it can be done in Java see `MlMappingsUpgradeIT`

**Comment by benwtrent** [comment]
@davidkyle I am not sure there is a reliable way to keep a DF in the `STARTED` state. Since we only support batch data frame transforms, we cannot keep the task around. we COULD add a bunch of data and HOPE that it does not all get processed so the data frame stays started, but that seems like it would be a very flaky test.

**Comment by benwtrent** [comment]
@davidkyle I am attempting to add some BWC tests that have a continuous data frame running while the upgrade is going...hopefully I get it figured out :)

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/checkpoint/DataFrameTransformsCheckpointService.java [hunk 1]
```diff
@@ -222,6 +224,16 @@ private long getTimeStampForTimeBasedSynchronization(SyncConfig syncConfig, long
         for (ShardStats shard : shards) {
             String indexName = shard.getShardRouting().getIndexName();
             if (userIndices.contains(indexName)) {
+                SeqNoStats seqNoStats = shard.getSeqNoStats();
+                // SeqNoStats could be `null`. This indicates that an `AlreadyClosed` exception was thrown somewhere down the stack
+                // Indicates that the index COULD be closed, or at least that the shard is not fully recovered yet.
+                if (seqNoStats == null) {
```
Comment (benwtrent) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/checkpoint/DataFrameTransformsCheckpointService.java]: @hendrikmuhs let me know what you think.

It is possible for this to throw an NPE, so to protect against that, I am still throwing a checkpoint exception with the index that failed, and that we were unable to gather `seq_no_stats`. The user making the call should not see any privileged information.

I opted to log the `ShardStats` as a warning. If that is too much information, we will have to change this to figure out how to log actionable information given the `seq_no_stats` being null.

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**